temporalio/client.py

```diff
@@ -595,6 +603,44 @@ def get_workflow_handle_for(
             result_type=defn.ret_type,
         )
 
+    async def list_workflows(
+        self,
+        query: Optional[str] = None,
+        *,
+        page_size: int = 1000,
+        next_page_token: Optional[bytes] = None,
+        rpc_metadata: Mapping[str, str] = {},
+        rpc_timeout: Optional[timedelta] = None,
+    ) -> WorkflowExecutionAsyncIterator:
+        """List workflows.
+
+        This eagerly requests the first page and will raise an exception if
+        not available.
+
+        Args:
+            query: A Temporal visibility list filter. See Temporal documentation
+                concerning visibility list filters including behavior when left
+                unset.
+            page_size: Number of results for each page.
+            next_page_token: A previously obtained next page token if doing
+                pagination.
```

Comment:
Why expose this in the high level API?

Reply:
Because people might want to paginate. Using this to drive a web/terminal UI is reasonable